www/src/components/repository/Repository.tsx

```diff
@@ -23,14 +28,23 @@ import { RepositorySideCar } from './RepositorySideCar'
 
 import { REPOSITORY_QUERY } from './queries'
 
+// To test whether the provided url param is a uuid or a name
+const uuidRegex = /^[0-9a-f]{8}-[0-9a-f]{4}-[0-5][0-9a-f]{3}-[089ab][0-9a-f]{3}-[0-9a-f]{12}$/i
+
 function Repository() {
   const { name } = useParams()
   const [searchParams] = useSearchParams()
-  const { data } = useQuery(REPOSITORY_QUERY, { variables: { name } })
+  const variables: any = { name }
+
+  if (name && uuidRegex.test(name)) {
+    variables.repositoryId = name
```

Comment:
this will technically keep the `name` attribute in the object, which probably works but it'd be safer to just do something like:

```js
variables = uuidRegex.test(name) ? { repositoryId: name} : { name}
```

(that way it doesn't rely on the precedence the api imposes to behave correctly)